codebundles/k8s-fluxcd-helm-health/runbook.robot

```diff
@@ -0,0 +1,125 @@
+Metadata            Display Name    Kubernetes FluxCD HelmRelease TaskSet
+Metadata            Supports    Kubernetes,AKS,EKS,GKE,OpenShift,FluxCD
+Library             RW.Core
+Library             RW.CLI
+Library             RW.Utils
+Library             RW.platform
+
+Suite Setup         Suite Initialization
+
+
+*** Tasks ***
+List all available FluxCD Helmreleases    
+    [Documentation]    List all FluxCD helmreleases that are visible to the kubeconfig.    
+    [Tags]        FluxCD     Helmrelease     Available    List
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} --context ${context}
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    Helmreleases available: \n ${stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Fetch All FluxCD Helmrelease Versions  
+    [Documentation]    List helmreleases and  the last attempted software version and the current running version.  
+    [Tags]        FluxCD     Helmrelease    Versions
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} -o=jsonpath="{range .items[*]}{'\\nName: '}{@.metadata.name}{'\\nlastAppliedRevision:'}{@.status.lastAppliedRevision}{'\\nlastAttemptedRevision:'}{@.status.lastAttemptedRevision}{'\\n---'}{end}" --context ${context} || true
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    Helmreleases status errors: \n ${stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Fetch Mismatched FluxCD HelmRelease Version
+    [Documentation]    List helmreleases and use jq to display any releases where the last attempted software revision doesn't match the current running revision. Requires jq.  
+    [Tags]        FluxCD     Helmrelease    Version    Mismatched    Unhealthy
+    ${stdout}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} ${NAMESPACE} -o json --context ${context} | jq -r '.items[] | select(.status.lastAppliedRevision!=.status.lastAttemptedRevision) | "Name: " + .metadata.name + " Last Attempted Version: " + .status.lastAttemptedRevision + " Last Applied Revision: " + .status.lastAppliedRevision'
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    RW.CLI.Parse Cli Output By Line
+    ...    rsp=${stdout}
+    ...    set_issue_title="FluxCD Helmrelease Version Mismatch"
+    ...    lines_like_regexp=".+"
+    ...    mismatched_helmrelease_versions__raise_issue_if_contains="Name:"
```

Comment:
You dont need to double-quote the inputs since robot is already passing them in as strings, otherwise it's actually getting "/"Name:/""

Reply:
Just realized I miss a key point: `mismatched_helmrelease_versions` would be the name of a capture group supplied by `lines_like_regexp`

From the linked example below:

`r'(?P<pod_name>[\\w-]+)\\s+[\\w//]+\\s+(?P<pod_status>\\w+)\\s+(?P<pod_restarts>[\\d]+)'`
Causes the `get pods` output to be split by columns, each column is given a name, and iterated through like rows of a database. 

single capture group: `r'(?P<mismatched_helmrelease_versions>[\\w-]+)` would create a capture group, and this makes the dynamic variable you can raise issues on `mismatched_helmrelease_versions`

Reply:
I do make `_line` accessible at all times and just represents the current line as a whole. If you want a simple contains check that should suffice. So:
`_line__raise_issue_if_contains="Name:"`

Reply:
This last approach is much better than the previous ones.

Reply:
I think my last commit should resolve this in a way that mostly makes sense to me (and works as tested against another codebundle I'm working on)